src/runtime/particle-execution-context.ts

```diff
@@ -259,7 +259,62 @@ export class ParticleExecutionContext {
   }
 
   private async reloadParticle(id: string) {
```

Comment:
Here's a challenge: Is there any space for unifying instantiateParticle() and reloadParticle() without obscuring these methods? There seem to be big sections of this methods that are very similar or almost the same.

Reply:
Updated! (I spliced the methods into three main parts).

Reply:
Well done @sherry-pra !